trino/base/params.env

```diff
@@ -0,0 +1,14 @@
+s3_endpoint_url=changeme
+s3_endpoint_url_prefix=https://
+s3_credentials_secret=aws-secret
+trino_environment=trino
+trino_db_secret=trino-db-secret
+hive_metastore_cpu_request=1
```

Comment:
I don't think the CPU requests needs to be this high. Keep in mind that request is reserved even if the pod is fully idle.

Reply:
I tend to disagree with that because this is just a default value for a parameter so user can set this value to what is better to their environments, and I can tell you that users might need to increase this value since this is a Java application and we don't have any evidences that Java behaves well with less than 1 CPU. It's too risky say that request can be 500m or less because of the JVM implementation.

Reply:
Interesting. Do you have any links to explanation or deepdives why a full CPU core has to be fully reserved for a Java pod?

I surely don't have as much experience with this as you do, but it doesn't seem right to me at all. Thriftserver in ODH has CPU request of 400m and is running just fine:
https://github.com/opendatahub-io/odh-manifests/blob/9e0f7842558b41098d89653c0b6f74a30cb8736d/thriftserver/thriftserver/base/thriftserver.yaml#L56

In idle it rests at 0.001 CPU for us, and there's no reason for it to reserve full CPU in that time either...

Reply:
There are many things we can discuss and I don't think we should start this discussion here, but let me give you an example: Let's say we use Serial Collector for the GC algorithm (and this is the (recommendation)[https://developers.redhat.com/blog/2017/04/04/openjdk-and-containers/] for 1 CPU), this collector thread will work concurrently with the application threads and thus running what we call a stop-the-world collection to free up the memory. We don't have any benchmarks with Hive Metastore working with 1GB of memory, but considering that 1GB is the total amount of memory reserved for the pod and JVM has other memory areas to manage, the Garbage Collection task can be frequent in this scenario and thus triggering a lot of GC tasks. Having that said, I'm not saying one or another configuration is wrong, but in order to find the best value for resource requests, we need a separate benchmark test to evaluate other variables.

Reply:
Hm, are you disputing memory? I'm discussing **CPU reservation** here, not memory.

I don't see recommended CPU requests in the article anywhere. I have no problem with the memory request, memory is usually very cheap on clusters... CPU is what I think is too high for a reservation. I can't find any proof that lower CPU requests may cause Java containers troubles, since the cluster allows seamless scaling up to the CPU limit if needed.